ipm_library/ipm_library/utils.py

```diff
@@ -14,10 +14,10 @@
 
 from typing import Optional, Tuple
 
-from builtin_interfaces.msg import Time
 from geometry_msgs.msg import Transform
 import numpy as np
 from rclpy.duration import Duration
+from rclpy.time import Time
```

Comment:
What type check did you use? Was it static or during runtime? Because I am quite sure the time used here is the time message. And the rclpy time has a time source etc. and this is not compatible.

Reply:
Static.

The types from TF said its this one

Reply:
But here we put it into a message and the message has this type.